selfdrive/loggerd/loggerd.cc

```diff
@@ -214,16 +204,28 @@ void encoder_thread(int cam_idx) {
         continue;
       }
 
-      //printf("logger latency to tsEof: %f\n", (double)(nanos_since_boot() - extra.timestamp_eof) / 1000000.0);
+      // Decide if we should rotate
+      pthread_mutex_lock(&s.rotate_lock);
+      s.last_camera_seen_tms = millis_since_boot();
 
-      // all the rotation stuff
+      bool new_segment = true;
+      for (auto &r : s.rotate_state) {
+        new_segment &= (((r.stream_frame_id >= r.last_rotate_frame_id + SEGMENT_LENGTH * MAIN_FPS) &&
+                        (!r.should_rotate) && (r.initialized)) ||
+                        (!r.enabled));
+#ifndef QCOM2
+          break; // only look at fcamera frame id if not QCOM2
+#endif
+      }
+
+      if (new_segment) trigger_rotate();
+      pthread_mutex_unlock(&s.rotate_lock);
+
+      // check if this encoder needs to rotate
       {
         pthread_mutex_lock(&s.rotate_lock);
```

Comment:
@deanlee Do your remember the purpose of this lock & unlock sequence?

Reply:
I don’t know why they are here too, seems unnecessary.

Reply:
ok, will remove then!

Reply:
Let me know if you have other comments on this PR!